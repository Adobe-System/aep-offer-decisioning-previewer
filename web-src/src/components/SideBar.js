/*
 * <license header>
 */

import React from "react";
import { NavLink } from "react-router-dom";

const SideBar = (props) => {
  return (
    <ul className="SideNav">
      <li className="SideNav-item">
        <NavLink
          className="SideNav-itemLink"
          activeClassName="is-selected"
          aria-current="page"
          exact
          to="/"
        >
          Home
        </NavLink>
      </li>
      {props.isSandboxSelected && (
        <li className="SideNav-item">
          <NavLink
            className="SideNav-itemLink"
            activeClassName="is-selected"
            aria-current="page"
            to="/previewer"
          >
            Previewer
          </NavLink>
        </li>
      )}
      <li className="SideNav-item">
        <NavLink
          className="SideNav-itemLink"
          activeClassName="is-selected"
          aria-current="page"
          to="/aiostatelist"
        >
          AIO State List
        </NavLink>
      </li>
      <li className="SideNav-item">
        <NavLink
          className="SideNav-itemLink"
          activeClassName="is-selected"
          aria-current="page"
          to="/about"
        >
          About this Apps
        </NavLink>
      </li>
    </ul>
  );
};

export default SideBar;
