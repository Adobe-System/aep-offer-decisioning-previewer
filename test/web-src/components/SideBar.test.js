import React from "react";
import { Provider, lightTheme } from "@adobe/react-spectrum";
import { render, screen, cleanup } from "@testing-library/react";
import { HashRouter as Router } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";
import mock from "../../actions/mock";
import SideBar from "../../../web-src/src/components/SideBar";

afterEach(cleanup);

describe("<SideBar> on load", () => {
  it("renders correctly ", () => {
    const { asFragment } = render(
      <Router>
        <Provider theme={lightTheme} colorScheme={`dark`}>
          <SideBar />
        </Provider>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("<SideBar> on sandbox selection", () => {
  it("renders correctly", () => {
    const fakeProps = { isSandboxSelected: true };
    const { asFragment } = render(
      <Router>
        <SideBar {...fakeProps} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
