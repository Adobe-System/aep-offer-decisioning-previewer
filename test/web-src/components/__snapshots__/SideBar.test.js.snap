// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar> on load renders correctly  1`] = `
<DocumentFragment>
  <div
    class="_spectrum_d43d2 _spectrum_60105 _spectrum--darkest_acf7f _spectrum--large_38267 _spectrum_5be2e _spectrum--medium_5be2e _spectrum--large_5be2e _spectrum--darkest_5be2e _spectrum--dark_5be2e _spectrum--light_5be2e _spectrum--lightest_5be2e"
    dir="ltr"
    lang="en-US"
    style="isolation: isolate; color-scheme: dark;"
  >
    <ul
      class="SideNav"
    >
      <li
        class="SideNav-item"
      >
        <a
          aria-current="page"
          class="SideNav-itemLink is-selected"
          href="#/"
        >
          Home
        </a>
      </li>
      <li
        class="SideNav-item"
      >
        <a
          class="SideNav-itemLink"
          href="#/about"
        >
          About this Apps
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<SideBar> on sandbox selection renders correctly 1`] = `
<DocumentFragment>
  <ul
    class="SideNav"
  >
    <li
      class="SideNav-item"
    >
      <a
        aria-current="page"
        class="SideNav-itemLink is-selected"
        href="#/"
      >
        Home
      </a>
    </li>
    <li
      class="SideNav-item"
    >
      <a
        class="SideNav-itemLink"
        href="#/previewer"
      >
        Previewer
      </a>
    </li>
    <li
      class="SideNav-item"
    >
      <a
        class="SideNav-itemLink"
        href="#/about"
      >
        About this Apps
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
